airflow/jobs/scheduler_job.py

```diff
@@ -1706,62 +1342,326 @@ def _run_scheduler_loop(self) -> None:
             loop_duration = loop_end_time - loop_start_time
             self.log.debug("Ran scheduling loop in %.2f seconds", loop_duration)
 
-            if not is_unit_test:
+            if not is_unit_test and not num_queued_tis and not num_finished_events:
+                # If the scheduler is doing things, don't sleep. This means when there is work to do, the
+                # scheduler will run "as quick as possible", but when it's stopped, it can sleep, dropping CPU
+                # usage when "idle"
                 time.sleep(self._processor_poll_interval)
 
+            if self.num_runs > 0 and loop_count >= self.num_runs:
```

Comment:
```suggestion
            if loop_count >= self.num_runs > 0:
```
WDYT?

Reply:
I find this a bit harder to work out what it's actually doing/checking.

Reply:
I'm afraid that pylint will require this anyway - we are still not running it over those files:
```
R1716: Simplify chained comparison between the operands (chained-comparison)
```

Reply:
Good enough for me then

Reply:
I am fine with both :)

Reply:
Done in https://github.com/apache/airflow/pull/10956/commits/7fa8828b0bffb597fc681583faf8ee031bb6c76f